File: src/gestorAplicacion/Consultorio.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package gestorAplicacion;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Map;
import java.util.TreeMap;
import java.util.ArrayList;

import gestorAplicacion.personas.*;

/**
 *
 * @author Roger Vera
 */
public class Consultorio implements Serializable{
	private static final long serialVersionUID = 1L;
    private Medico medico;
    private Paciente paciente;
    private String tipoConsultorio;
    private final int id;
    private boolean disponibilidad;
    public Map<LocalDateTime, Consulta> consultas = new TreeMap<LocalDateTime, Consulta>();
    public static ArrayList<Consultorio> consultorios = new ArrayList<Consultorio>();
    
    //constructor

    public Consultorio(Medico medico, Paciente paciente, String tipoConsultorio, boolean disponibilidad, LocalDateTime fecha, Consulta consulta) {
        this.medico = medico;
        this.paciente = paciente;
        this.tipoConsultorio = tipoConsultorio;
        this.id = (int)(Math.random()*10000+1);
        this.disponibilidad = disponibilidad;
        consultorios.add(this);
        consultas.put(fecha, consulta);
    }
       
  
        
    public Consultorio(Medico medico, Paciente paciente, String tipoConsultorio, boolean disponibilidad) {
    	this.medico = medico;
        this.paciente = paciente;
        this.tipoConsultorio = tipoConsultorio;
        this.disponibilidad = disponibilidad;
    	this.id = (int)(Math.random()*10000+1);
    	consultorios.add(this);
	}

    //getter & setter     
    
	public Medico getMedico() {
        return medico;
    }

    public void setMedico(Medico medico) {
        this.medico = medico;
    }

    public Paciente getPaciente() {
        return paciente;
    }

    public void setPaciente(Paciente paciente) {
        this.paciente = paciente;
    }

    public String getTipoConsultorio() {
        return tipoConsultorio;
    }

    public void setTipoConsultorio(String tipoConsultorio) {
        this.tipoConsultorio = tipoConsultorio;
    }

	public int getId() {
		return id;
	}


	public Map<LocalDateTime, Consulta> getConsultas() {
		return consultas;
	}


	public void setConsultas(Map<LocalDateTime, Consulta> consultas) {
		this.consultas = consultas;
	}


	public boolean isDisponibilidad() {
		return disponibilidad;
	}


	public void setDisponibilidad(boolean disponibilidad) {
		this.disponibilidad = disponibilidad;
	}
   
    //metodos
    
    
    
}

File: practica2-grupo01-Equipo09/src/gestorAplicacion/Consultorio.py
"""
La clase Consultorio es utilizada para almacenar los atributos y aplciar cambios a los consultorios del sistema.
Se tienen atributos como Id, el tipo de consultorio y un diccionario de las consultas que el consultorio tiene asignadas
en determinadas fechas.
"""
from gestorAplicacion.personas.Administrador import Administrador
import random as rd
class Consultorio():
    consultorios=[]
    def __init__(self, id = int(rd.random() * 10000 + 1), medico = None, paciente = None, tipoConsultorio = None, 
                adquirido = False, disponibilidad = True, consultas = None, costo = 100000):
        if consultas is None:
            consultas = {}
        self._id = id
        self._medico = medico
        self._paciente = paciente
        self._tipoConsultorio = tipoConsultorio
        self._disponibilidad = disponibilidad
        self._consultas = consultas
        self._adquirido = adquirido
        self._costo = costo
        Administrador.consultorios.append(self)
    
    # Getters y Setters
    def getId(self):
        return self._id
    
    def setId(self, id):
        self._id = id
    
    def getMedico(self):
        return self._medico
    
    def setMedico(self, medico):
        self._medico = medico
    
    def getPaciente(self):
        return self._paciente
    
    def setPaciente(self, paciente):
        self._paciente = paciente
    
    def getTipoConsultorio(self):
        return self._tipoConsultorio
    
    def setTipoConsultorio(self, tipoConsultorio):
        self._tipoConsultorio = tipoConsultorio
    
    def isDisponibilidad(self):
        return self._disponibilidad
    
    def setDisponibilidad(self, disponibilidad):
        self._disponibilidad = disponibilidad
        
    def getConsultas(self):
        return self._consultas
    
    def setConsultas(self, consultas):
        self._consultas = consultas
        
    def isAdquirido(self):
        return self._adquirido
    
    def setAdquirido(self, adquirido):
        self._adquirido =adquirido
    
    def getCosto(self):
        return self._costo
    
    def setCosto(self, costo):
        self._costo = costo
    
        
            